src/shared_modules/rsync/include/rsync.hpp

```diff
@@ -26,30 +26,34 @@
 #endif
 
 #include <functional>
+#include <thread>
 #include "json.hpp"
 #include "commonDefs.h"
 
 using SyncCallbackData = const std::function<void(const std::string&)>;
 
-class EXPORTED RemoteSync 
+class EXPORTED RemoteSync
 {
 public:
     /**
     * @brief Initializes the shared library.
     *
-    * @param logFunction pointer to log function to be used by the rsync.
+    * @param logFunction Pointer to log function to be used by the rsync.
     */
     static void initialize(std::function<void(const std::string&)> logFunction);
 
     /**
-     * @brief Remote sync initializes the instance. 
+     * @brief Remote sync initializes the instance.
+     *
+     * @param threadPoolSize Size of the thread pool.
+     * @param maxQueueSize Maximum size of the queue, if the value is 0, it is unlimited.
      */
-    RemoteSync();
+    RemoteSync(const unsigned int threadPoolSize = std::thread::hardware_concurrency(), const size_t maxQueueSize = 0);
```

Comment:
Hi @Dwordcito

Maybe the `UNLIMITED_QUEUE_SIZE` macro could be moved to a common file and used here as well.